### PR TITLE
vc: do not follow symlink when umounting contanier host path

### LINKED ISSUE #2474: harden container hostpath cleanup
# Description of problem

A container's host path might be changed by the guest to point to some other places by placing a symlink there. kata runtime should not follow link when unmounting them otherwise we might end up unmounting some other mountpoints unexpectedly.

### PR DESCRIPTION
**Pull request by bergwolf**
So that if a guest changes it, we do not end up
propagating the error.

Fixes: #2474

### CODE REVIEW

**Review by devimc — APPROVED** [review]

**Review by lifupan — APPROVED** [review]

File: virtcontainers/mount.go [hunk 1]
```diff
@@ -277,6 +281,10 @@ func bindMount(ctx context.Context, source, destination string, readonly bool) e
 		return fmt.Errorf("Could not create destination mount point %v: %v", destination, err)
 	}
 
+	if dest, err := filepath.EvalSymlinks(destination); err == nil && !strings.HasPrefix(dest, filepath.Clean(topDir)) {
+		return fmt.Errorf("Invalid mount destination %s must not escape", destination)
+	}
+
```
Comment (laijs) [inline virtcontainers/mount.go]: there is a "TOCTOU" here

Comment (bergwolf) [reply virtcontainers/mount.go]: I dropped it as it also breaks vm template.

**Pull request merged**